asserts/database.go

```diff
@@ -291,12 +291,18 @@ func (db *Database) Check(assert Assertion) error {
 // It will return an error when trying to add an older revision of the assertion than the one currently stored.
 func (db *Database) Add(assert Assertion) error {
 	assertType := assert.Type()
+
+	keyLen := len(assertType.PrimaryKey)
+	if keyLen == 0 {
+		return fmt.Errorf("cannot add %q assertion with zero-length primary key", assertType.Name)
```

Comment:
Message seems slightly confusing. Isn't this more like:
`internal error: assertion type %q has no primary key`
?